### DISCUSSION TITLE
Event Listeners don't register if the element isn't wrapped in `x-data="{}"`

### DISCUSSION THREAD

**Discussion by KonnorRogers** [alpinejs/alpine#4273] [7. Bugs (archived)] [answered]
I don't know if im holding it wrong and just missed it in the docs, but it seems like event listeners don't bind unless they are either wrapped by an element with `x-data` or if the element itself has `x-data`.

Codepen: https://codepen.io/paramagicdev/pen/NWVzGgr

**Comment by KonnorRogers** [comment]
Seems i am holding it wrong.

https://alpinejs.dev/directives/data

**Comment by ekwoka** [comment, marked as answer]
That's expected behavior.

**Reply by KonnorRogers** [reply]
@ekwoka yup, I realized that as I read through the docs more, which is why I closed this  🙈

**Reply by ekwoka** [reply]
Yup, just wanted to reply in case someone else ends up here, to see.
